src/OrleansAzureUtils/Hosting/ServiceRuntimeWrapper.cs

```diff
@@ -183,11 +183,9 @@ private void Initialize()
                 logger.Warn(ErrorCode.AzureServiceRuntime_NotLoaded, msg1);
 
                 // Microsoft.WindowsAzure.ServiceRuntime isn't loaded. We may be running within a web role or not in Azure.
-                // Trying to load by partial name, so that we are not version specific.
-                // Assembly.LoadWithPartialName has been deprecated. Is there a better way to load any version of a known assembly?
-                #pragma warning disable 618
-                assembly = Assembly.LoadWithPartialName("Microsoft.WindowsAzure.ServiceRuntime");
-                #pragma warning restore 618
+#pragma warning disable 618
+                assembly = Assembly.Load(new AssemblyName("Microsoft.WindowsAzure.ServiceRuntime"));
+#pragma warning restore 618
```

Comment:
hmm... how do you feel like using.. 

```
var roleRootDir = Environment.GetEnvironmentVariable("RoleRoot");
string osMode = "x64";
 if (!Environment.Is64BitOperatingSystem)
        osMode = "x86";
assembly = Assembly.LoadFrom(roleRootDir + "\\base\\" + osMode + "\\Microsoft.WindowsAzure.ServiceRuntime.dll"); 
```

I tried to look into queryAssemblyInfo from GAC, looks like it only support query with full assembly name also.